### ISSUE TITLE
Bug in coast -JW-65

### ISSUE THREAD

**Issue by Esteban82** [GenericMappingTools/gmt#1982]
When I plot a map centered in -65 (-JW-65/15c) some countries in the far east are painted in black. It do not occurs when I the map is center in the greenwich meridian (-JW15c) or I use the classic mode.

[Modern_Mapamundi_Bug_W0.pdf](https://github.com/GenericMappingTools/gmt/files/3806187/Modern_Mapamundi_Bug_W0.pdf)
[Modern_Mapamundi_Bug_W-65.pdf](https://github.com/GenericMappingTools/gmt/files/3806188/Modern_Mapamundi_Bug_W-65.pdf)

```
gmt begin Modern_Mapamundi_Bug_W-65
	gmt coast -Rd -JW-65/15c -Sdodgerblue2
gmt end show
```

**System information**

* Operating system: Windows 10
* GMT version (`gmt --6.0.0 64-bit version`):

**Comment by PaulWessel** [comment]
I can confirm that problem on macOS as well and will have a look tomorrow.

**Comment by PaulWessel** [comment]
Some more research shows that the same problem is present in classic mode (as long as you use the same resolution, i.e. -Dl) and this goes back to GMT 4 at least.  So as far as I can tell a long-lived problem.  Will need to see if it is a GSHHG problem versus a coding problem.

**Comment by PaulWessel** [comment]
I get the same problem with gshhg 2.3.0.  It depends on the center meridian in the sense that the problem applies to polygon pieces near the right (east) boundary, but a we change the meridian there are other polygons that are close and turn black. So that suggests it is related to decisions made after projecting the coordinates or something with the east periodic boundary.  It has the same problem with -JH, -JN, etc but not -JQ (straight boundary).

**Comment by PaulWessel** [comment]
It also applies to all resolutions but is bin-based, so maybe only affecting the polygons of the far-east bin along the periodic boundary.

**Comment by PaulWessel** [comment]
I removed modern mode from the title since it is unrelated to mode.  These two commands give the exact same results:

```
gmt coast -Rd -JW-45/15c -Sdodgerblue2 -Dl -ps modern
gmt pscoast -Rd -JW-45/15c -Sdodgerblue2 -Dl > classic.ps
```

**Comment by PaulWessel** [comment]
Fixed by #1995.
